examples/slice/TcpFallback/Client/Program.cs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) ZeroC, Inc.
+
+using IceRpc;
+using IceRpc.Transports.Quic;
+using Microsoft.Extensions.Logging;
+using System.Net.Security;
+using System.Security.Cryptography.X509Certificates;
+using VisitorCenter;
+
+using ILoggerFactory loggerFactory = LoggerFactory.Create(builder =>
+    builder
+        .AddSimpleConsole()
+        .AddFilter("IceRpc", LogLevel.Information));
+
+// SSL setup
+using var rootCA = new X509Certificate2("../../../../certs/cacert.der");
+var clientAuthenticationOptions = new SslClientAuthenticationOptions
+{
+    RemoteCertificateValidationCallback = (sender, certificate, chain, errors) =>
+    {
+        using var customChain = new X509Chain();
+        customChain.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
+        customChain.ChainPolicy.DisableCertificateDownloads = true;
+        customChain.ChainPolicy.TrustMode = X509ChainTrustMode.CustomRootTrust;
+        customChain.ChainPolicy.CustomTrustStore.Add(rootCA);
+        return customChain.Build((X509Certificate2)certificate!);
+    }
+};
+
+// Use our own factory method (see below) to create a client connection using QUIC with a fallback to TCP.
+await using ClientConnection connection = await CreateClientConnectionAsync(
+    new Uri("icerpc://localhost"),
+    clientAuthenticationOptions,
+    loggerFactory.CreateLogger<ClientConnection>());
+
+// Create an invocation pipeline and install the logger interceptor.
+Pipeline pipeline = new Pipeline()
+    .UseLogger(loggerFactory)
+    .Into(connection);
+
+// Create a greeter proxy with this invocation pipeline.
+var greeter = new GreeterProxy(pipeline);
+
+string greeting = await greeter.GreetAsync(Environment.UserName);
+
+Console.WriteLine(greeting);
+
+await connection.ShutdownAsync();
+
+// Creates a client connection connected to the server over QUIC if possible. If the QUIC connection establishment
+// fails, fall back to a TCP connection. The caller must dispose the returned connection.
+static async Task<ClientConnection> CreateClientConnectionAsync(
+    Uri serverAddressUri,
+    SslClientAuthenticationOptions clientAuthenticationOptions,
+    ILogger logger)
+{
+#pragma warning disable CA2000 // Dispose objects before losing scope
```

Comment:
I tried but could not find a better solution than to suppress this warning.

Reply:
One way to get rid of the warning is use try/finally as mentioned in CA2000 help

```
ClientConnection? connection = null;
ClientConnection? tmp = null;
try
{
    tmp = new ClientConnection(
       serverAddressUri,
       clientAuthenticationOptions,
       multiplexedClientTransport: new QuicClientTransport(),
       logger: logger);
    await tmp.ConnectAsync();
    connection = tmp;
    tmp = null;
}
catch (Exception exception)
{
    logger.LogInformation(exception, "Failed to connect to server using QUIC, falling back to TCP");
}
finally
{
    if (tmp is not null)
    {
        await tmp.DisposeAsync();
    }
}

return connection is null ?
    new ClientConnection(serverAddressUri, clientAuthenticationOptions, logger: logger) : connection;
```

Reply:
I don't find this code very elegant / readable.